qa/integration-tests/src/test/java/io/camunda/it/client/SearchUserTaskTest.java

```diff
@@ -0,0 +1,481 @@
+
+    calendar.add(Calendar.DAY_OF_YEAR, 3);
+    String dayAfter = dateFormat.format(calendar.getTime());
+
+    // Create a DateFilter with the formatted date strings for the range
+    final DateFilter dateFilter = new DateFilter().from(dayBefore).to(dayAfter);
+
+    final var result =
+        camundaClient.newUserTaskQuery().filter(f -> f.userTaskDueDate(dateFilter)).send().join();
+
+    assertThat(result.items().size()).isEqualTo(3);
+
+    // Check if the completion date is without the range
+    calendar.add(Calendar.DAY_OF_YEAR, -3);
+    dayBefore = dateFormat.format(calendar.getTime());
+
+    // Reset to current time
+    calendar.setTime(now);
+
+    // Calculate the date one day after
+    calendar.add(Calendar.DAY_OF_YEAR, 1);
+    dayAfter = dateFormat.format(calendar.getTime());
+
+    final DateFilter dateFilterOutOfRange = new DateFilter().from(dayBefore).to(dayAfter);
+    final var resultOutOfRange =
+        camundaClient
+            .newUserTaskQuery()
+            .filter(f -> f.userTaskDueDate(dateFilterOutOfRange))
+            .send()
+            .join();
+    assertThat(resultOutOfRange.items().size()).isEqualTo(0);
+  }
+
+  @Test
+  public void shouldRetrieveTaskByFollowUpDate() {
+    final SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+    dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+
+    final Date now = new Date();
+
+    final Calendar calendar = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+    calendar.setTime(now);
+    calendar.add(Calendar.DAY_OF_YEAR, -4);
+    String dayBefore = dateFormat.format(calendar.getTime());
+
+    calendar.setTime(now);
+
+    calendar.add(Calendar.DAY_OF_YEAR, 3);
+    String dayAfter = dateFormat.format(calendar.getTime());
+
+    // Create a DateFilter with the formatted date strings for the range
+    final DateFilter dateFilter = new DateFilter().from(dayBefore).to(dayAfter);
+
+    final var result =
+        camundaClient
+            .newUserTaskQuery()
+            .filter(f -> f.userTaskFollowUpDate(dateFilter))
+            .send()
+            .join();
+
+    assertThat(result.items().size()).isEqualTo(3);
+
+    // Check if the completion date is without the range
+    calendar.add(Calendar.DAY_OF_YEAR, -3);
+    dayBefore = dateFormat.format(calendar.getTime());
+
+    // Reset to current time
+    calendar.setTime(now);
+
+    // Calculate the date one day after
+    calendar.add(Calendar.DAY_OF_YEAR, 1);
+    dayAfter = dateFormat.format(calendar.getTime());
+
+    final DateFilter dateFilterOutOfRange = new DateFilter().from(dayBefore).to(dayAfter);
+    final var resultOutOfRange =
+        camundaClient
+            .newUserTaskQuery()
+            .filter(f -> f.userTaskFollowUpDate(dateFilterOutOfRange))
+            .send()
+            .join();
+    assertThat(resultOutOfRange.items().size()).isEqualTo(0);
+  }
+
+  @Test
+  public void shouldThrowAnExceptionWhenSearchBeforeAndAfterAreApplied() {
```

Comment:
🔧 This can already be tested in the client itself, since it's just input validation. This doesn't have to be an integration test which simply is more costly due to its setup.